File: FulltimeProgrammeren2/module7/sentimental/vigenere.py
 #**************************************************************************************
 # vigenere.py
 #
 # Fulltime Programmeren 2
 # Robin Laponder
 #
 # Encrypts messages using Vigenère’s cipher.
 #**************************************************************************************

from sys import argv
from cs50 import get_string

def main():
    # check if user typed in one argument
    if len(argv) != 2 or not argv[1].isalpha():
        print("Usage: python vigenere.py k")
        exit(1)

    # convert entered argument to keyword
    keyword = argv[1].lower()

    # prompt user to enter plaintext
    plaintext = get_string("plaintext: ")

    # print ciphertext
    print("ciphertext: ", end = "")
    c = 0
    for i in plaintext:
        key = get_key(keyword, c)

        # Character is uppercase
        if i.isupper():
            upper = (((ord(i) - ord('A')) + key) % 26) + ord('A')
            print(chr(upper), end="")
            c += 1

        # Character is lowercase
        elif i.islower():
            lower = (((ord(i) - ord('a')) + key) % 26) + ord('a')
            print(chr(lower), end="")
            c += 1

        else:
            print("{}".format(i), end="")

    # print a new line at the end
    print()

# get key for every letter
def get_key(keyword, counter):
    letter = counter % len(keyword)
    return ord(keyword[letter]) - 97

if __name__ == "__main__":
    main()

File: FulltimeProgrammeren1/module2/vigenere.c
/****************************************************************************
 * viginere.c
 *
 * Fulltime Programmeren 1
 * Robin Laponder
 *
 * Encrypts messages using Vigenère’s cipher.
 ***************************************************************************/
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
// 26 is the number of chars in the alfabet
#define LENGTH_ALFABET 26

    // check if there is one argument
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    // variables
    string key = argv[1];
    int keylength = strlen(key);

    // check if argument contains of only letters
    for (int i = 0; i < keylength; i++)
    {
        if (!isalpha(key[i]))
        {
            printf("Usage: ./viginere k\n");
            return 1;
        }
    }

    // prompt user to enter text
    string plaintext = get_string("plaintext: ");

    // scan for lowercase an uppercase letters and print in ciphertext
    printf("ciphertext: ");
    for (int i = 0, j = 0, n = strlen(plaintext); i < n; i++, j++)
    {
        // convert key to lowercase letters
        int keyletter = tolower(key[j % keylength]) - 'a';

        // scan for lowercase letters and print encrypted characters
        if (islower(plaintext[i]))
        {
            printf("%c", 'a' + (plaintext[i] - 'a' + keyletter) % LENGTH_ALFABET);
        }

        // scan for uppercase letters and print encrypted characters
        else if (isupper(plaintext[i]))
        {
            printf("%c", 'A' + (plaintext[i] - 'A' + keyletter) % LENGTH_ALFABET);
        }

        // if character is not a letter, print it
        else
        {
            printf("%c", plaintext[i]);
            j--;
        }
    }
    printf("\n");
    return 0;
}